lib/simplifyExpression/arithmeticSearch/index.js

```diff
@@ -1,8 +1,96 @@
 const ChangeTypes = require('../../ChangeTypes');
-const evaluate = require('../../util/evaluate');
+const clone = require('../../util/clone');
+const evaluate = require('math-evaluator').default;
 const Node = require('../../node');
 const TreeSearch = require('../../TreeSearch');
 
+const {build, query} = require('math-nodes');
+const {defineRule, rewriteNode} = require('math-rules');
+const {parse} = require('math-parser');
+
+const getRanges = (args, predicate) => {
+  const ranges = [];
+  let i;
+  let start = -1;
+  for (i = 0; i < args.length; i++) {
+    if (predicate(args[i])) {
+      if (start === -1) {
+        start = i;
+      }
+    }
+    else {
+      if (start !== -1 && i - start > 1) {
+        ranges.push([start, i]);
+      }
+      start = -1;
+    }
+  }
+  if (start !== -1 && i - start > 1) {
+    ranges.push([start, i]);
+  }
+  return ranges;
+};
+
+function applyRule(node, rule, changeType, reverse) {
+  let newNode = rewriteNode(rule, node);
+  if (!newNode) {
+    newNode = rewriteNode(reverse, node);
+  }
+  if (newNode) {
+    return Node.Status.nodeChanged(
+      changeType, node, newNode);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// ARITHMETIC
+
+// e.g. 2 + 2 -> 4 or 2 * 2 -> 4
+// TODO(kevinb): handle fractions
+const SIMPLIFY_ARITHMETIC = defineRule(
+  node => {
+    if (query.isOperation(node)) {
+      if (query.isAdd(node) || query.isMul(node)) {
+        if (node.args.every(query.isNumber)) {
+          return {node};
+        }
+        else {
+          const ranges = getRanges(node.args, query.isNumber);
+          if (ranges.length > 0) {
+            // For now we're only using the first range, but we'll
+            // want to use all ranges when we're applying a rule
+            // multiple times in the future.
+            const indexes = {
+              start: ranges[0][0],
+              end: ranges[0][1],
+            };
+            return {node, indexes};
+          }
+        }
+      }
+      else if (node.args.every(query.isNumber)) {
+        return {node};
+      }
+    }
+    return null;
+  },
+  // TODO: replace this with '#eval(#a)'
```

Comment:
that seems like a pretty important case - I don't think TreeSearch won't handle because it searches a node and then each of its children separately, not a subset of its children together